Feature/TS GOLD

gnguyen87 opened the pull request:
Add a new bundle for [TS-GOLD assessment](https://teachingstrategies.com/the-objectives-for-development-and-learning/).

jalvord1 committed:
initial commit for TS Gold

Commit:
Add seeds

Commit:
add more assessmentId

Commit:
Add transformations for studentAssessment

Commit:
add new gradeleveldescriptors for new assessments

Commit:
edit studentAssessment template

Commit:
remove objectAssessment

Commit:
refactor to take care of empty scoreResults and performanceLevels

Commit:
add languageDescriptor to assessment

Commit:
remove objectAssessments

Commit:
edit performanceLevels in assessment template

Commit:
add some docs to clarify design choices

jalvord1 commented on assessments/GOLD/earthmover.yaml:
```diff
@@ -0,0 +1,176 @@
+%}
+
+{% set school_years = [
+  'Fall 2022/2023'
```
We are going to have other files that have different values here (both for season & year), is there a way we could make this dynamic?

gnguyen87 replied:
I have refactored the code to make this and the below `admin_date` to be more dynamic. However, its validity depends on how we are assuming source files are managed.

jalvord1 commented on assessments/GOLD/earthmover.yaml:
```diff
@@ -0,0 +1,176 @@
+          namespace: "uri://www.teachingstrategies.com/assessment/gold"
+      - operation: modify_columns
+        columns:
+          admin_date: "{%raw%}{% if admin_date != '' %}{{value}}{% else %}{% if school_year == 'Fall 2022/2023'%}10/1/2022{% endif %}{% endif %}{%endraw%}"
```
We'll probably need to figure out how to make this dynamic as well

jalvord1 commented on assessments/GOLD/earthmover.yaml:
```diff
@@ -0,0 +1,176 @@
+          - admin_date
+          - assessment_id
+          - ChildID
+        new_column: student_assessment_id
```
Have we confirmed this convo of columns = uniqueness? My assumption is yes, just double checking!

gnguyen87 replied:
I believe so !

jalvord1 commented on assessments/GOLD/seeds/assessments.csv:
```diff
@@ -0,0 +1,12 @@
+assessmentIdentifier,assessmentTitle,namespace,assessmentCategoryDescriptor,academicSubjectDescriptor, languageDescriptor
+GOLD_SE,Social-Emotional,uri://www.teachingstrategies.com/assessment/gold,uri://ed-fi.org/AssessmentCategoryDescriptor#Early Learning - Social and emotional development,uri://ed-fi.org/AcademicSubjectDescriptor#Other,uri://ed-fi.org/LanguageDescriptor#eng
```
I used a slightly different namespace: uri://teachingstrategies.com/product/gold/

jalvord1 commented on assessments/GOLD/seeds/performanceLevelDescriptors.csv:
```diff
@@ -0,0 +1,6 @@
+codeValue,description,namespace,shortDescription
+Below,Student is below expectations,uri://www.teachingstrategies.com/assessment/PerformanceLevelDescriptor,Below
```
namespaces here also slightly different

Commit:
update namespace

Commit:
Make school_year a required CLI parameter + reconfigure admin_date to allow for dynamic usage

Commit:
testing new earthmover edits

Commit:
Refactor so that school_year can be dynamically created based on the column names themselves

gnguyen87 committed:
Update README.md

alchenist committed:
Add assessmentFamily to GOLD

alchenist committed:
Remove trailing slash in namespace for GOLD

alchenist committed:
Make namespace consistent between seeds and earthmover.yaml for GOLD

alchenist committed:
Fix academic subject descriptor format in GOLD templates

alchenist committed:
Fix Physical, Health, and Safety Education descriptor in GOLD seeds

alchenist committed:
Fix studentAssessments output for GOLD

- Include WHE in performance levels as intended
- Fix school year format (previously like 'Fall 2022/2023', now integer as Ed-Fi expects)
- Fix performance level format to match Ed-Fi format

alchenist committed:
Add whenAssessedGradeLevelDescriptor to GOLD studentAssessments

Maps original grade levels seen in SCDE exports to grade level
descriptors. Kindergarten is included as well to match current
anonymized sample file.

Refactors how grade levels are mapped to assessments as well; seed
file no longer has entries for each assessment ID and grade levels
are now cross joined with assessments using a dummy key.

alchenist committed:
fixup! Fix studentAssessments output for GOLD

alchenist committed:
Add filter for non-blank scores/performance in GOLD

alchenist committed:
Replace hardcoded Ed-Fi namespace with descriptor_namespace in GOLD

alchenist committed:
Move score and perf level logic to top of student assessments template for GOLD

alchenist committed:
Add longer descriptions for some GOLD assessment reporting method descriptors

alchenist committed:
Replace student ID references to make compatible with xwalking in GOLD

alchenist committed:
Implement student assessment ed org associations for GOLD

alchenist committed:
Remove debugging fragments from GOLD

alchenist committed:
GOLD: Factor out assessment category descriptors in case we need to send them in vendor namespace

alchenist committed:
Replace map_values with composition-friendly join

Commit:
update seeds

Commit:
refactor to make gold runway ready

Commit:
add metadata file (no need for SY input param bc this field is dynamically pulled from input file

Commit:
add descriptor_mapping_files

theokaufman committed:
rename metadata file

theokaufman committed:
Update README

theokaufman committed:
update templates

theokaufman committed:
update sample file for testing

theokaufman committed:
amend column rename for clarity

theokaufman committed:
Merge branch 'main' into feature/ts_gold
merging main in

theokaufman committed:
add correct academicSubjectDescriptor

theokaufman committed:
Update site id

theokaufman committed:
Trim values

theokaufman committed:
update registry
